core/src/ser.rs

```diff
@@ -561,6 +568,109 @@ impl<'a> Reader for StreamingReader<'a> {
 	}
 }
 
+/// Protocol version-aware wrapper around a `Buf` impl
```

Comment:
Does this actually need to be "protocol version aware"? I suspect we can get away with simplifying this.

Reply:
The reason for this is that the `Reader` trait has a method to access the protocol version. It is used in the protocol handler to convert from raw bytes to the structs, as deserializing these structs can be protocol version dependent.

Reply:
Ahh right. That was my fault... :+1: